esp/platform/application_config.cpp

```diff
@@ -295,7 +295,7 @@ bool addProtocol(IPropertyTree *legacyEsp, IPropertyTree *app)
 void addServices(IPropertyTree *legacyEsp, IPropertyTree *appEsp, const char *application, const char *auth, bool tls)
 {
     Owned<IPropertyTreeIterator> services = appEsp->getElements("application/services");
-    int port = appEsp->getPropInt("@port", 8880);
+    int port = appEsp->getPropInt("service/@port", appEsp->getPropInt("@port", 8880));
```

Comment:
I think these application yaml's are part of the image e.g. /opt/HPCCSystems/componentfiles/applications/eclwatch/

I guess picking up the legacy "@port" as a default here, ensures still works if the used with an older image, that also had @port set differently in it's image copy of esp.yaml.
But I don't think that can/has happened in reality.

Perhaps would be clearer not to have code pick up legacy "@port" here.

Reply:
Those application yamls can be thought of as ESPs "default.yaml".  The values can still be provided in values.yaml which needs this backward compatibility to work with any existing ones.

Reply:
if this code is used in bare-metal and it's possible it could pick up a old legacy format.
In cloud the schema will not allow the legacy spec.